src/OS8104A.ts

```diff
@@ -0,0 +1,697 @@
+    allocTimeout: ReturnType<typeof setTimeout> | null
+    streamAllocTimeout: ReturnType<typeof setTimeout> | null
+    delayTimer: ReturnType<typeof setTimeout> | null
+    getSourceTimeout: ReturnType<typeof setTimeout> | null
+    allocCheck: ReturnType<typeof setInterval> | null
+    lockInterval: ReturnType<typeof setInterval> | null
+    multiPartMessage: MostMessage
+    multiPartSequence: number
+    transceiverLocked: boolean
+
+    constructor(nodeAddress: number, groupAddress: number, freq: number) {
+        super()
+        this.spi = spi.openSync(0, 0, options)
+        this.interrupt = new Gpio(5, "in", "falling")
+        // TODO this had an unnoticed type error for debounce, now TS has saved the day, it may mess things up
+        // now that it's actually working
+        this.fault = new Gpio(6, "in", "both", { debounceTimeout: 50 })
+        this.status = new Gpio(16, "in", "both", { debounceTimeout: 50 })
+        this.mostStatus = new Gpio(26, "in", "both", { debounceTimeout: 10 })
+        this.reset = new Gpio(17, "out")
+        this.freq = freq
+        // TODO not sure why these were buffers, need to review
+        this.nodeAddressBuf = Buffer.alloc(2)
+        this.nodeAddressBuf.writeUint16BE(nodeAddress)
+        this.groupAddressBuf = Buffer.alloc(1)
+        this.groupAddressBuf.writeUInt8(groupAddress)
+        this.awaitAlloc = false
+        // REVIEW - is this a better method than just allowing the type to be a type or null?
+        this.allocResult = {
+            loc1: -1,
+            loc2: -1,
+            loc3: -1,
+            loc4: -1,
+            cl: -1,
+            answer1: "",
+            freeChannels: -1
+        }
+        this.awaitGetSource = false
+        this.getSourceResult = null
+        this.allocTimeout = null
+        this.streamAllocTimeout = null
+        this.getSourceTimeout = null
+        this.allocCheck = null
+        this.lockInterval = null
+        this.delayTimer = null
+        // REVIEW - is this a better method than just allowing the type to be a type or null?
+        this.multiPartMessage = {
+            targetAddressHigh: 0,
+            targetAddressLow: 0,
+            fBlockID: 0,
+            instanceID: 0,
+            fktId: 0,
+            opType: 0,
+            data: []
+        }
+        this.multiPartSequence = 0
+        this.transceiverLocked = true
+        this.startUp()
+        this.getRegisterConfig = getRegisterConfig
+
+        this.fault.watch((err, val) => {
+            if (err !== null && err !== undefined) {
+                throw err
+            }
+            console.log("fault", val)
+        })
+
+        this.status.watch((err, val) => {
+            if (err !== null && err !== undefined) {
+                throw err
+            }
+            console.log("status", val)
+        })
+
+        this.mostStatus.watch((err, val) => {
+            if (err !== null && err !== undefined) {
+                throw err
+            }
```

Comment:
Unsure on this, it seems the type inferred from onoff library is `Error | null | undefined` eslint seems to complain on a check of `if(err)` so again more research needed

Reply:
hmm - this should work just fine - perhaps with the tsconfig and updated eslint rules it wont complain

Reply:
You were right, updated both now it works fine

https://github.com/rhysmorgan134/SocketMost/blob/9ce8ed6645f6af46f293c8d8e9eb070ae297161b/src/driver/OS8104A.ts#L74-L104